src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -235,6 +235,7 @@ private static bool PlatformAnalysisAllowed(AnalyzerOptions options, Compilation
                     var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
                         cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
                         context.Options, SupportedOsRule, performValueContentAnalysis,
+                        pessimisticAnalysis: false,
```

Comment:
This is the core change needed to fix the bug. @buyaa-n would you be able to help validating if this fixes the aspnet core repo issue that @pranavkm was hitting by building locally with this change?

Reply:
Sure, i can validate it

Reply:
Yes, it was fixed with this update, thanks @mavasani